inlong-sort/sort-connectors/jdbc/src/main/java/org/apache/inlong/sort/jdbc/internal/JdbcBatchingOutputFormat.java

```diff
@@ -277,7 +294,7 @@ public final synchronized void writeRecord(In record) {
             if (executionOptions.getBatchSize() > 0
                     && batchCount >= executionOptions.getBatchSize()) {
                 flush();
-                if (sinkMetricData != null) {
+                if (sinkMetricData != null && dirtySink == null) {
                     sinkMetricData.invoke(rowSize, dataSize);
```

Comment:
why add dirtySink == null?

Reply:
if dirty sink is not null, the tablemetricExecutor will add metrics one time if it is initialized successfully. So the metric will be calculated twice.

Reply:
> 

@Yizhou-Yang It will exec this logic when I don't use dirtySink. Other logic report metric when I use dirtySink?

Reply:
yes

Reply:
> yes, TableMetricStatementExecutor will calculate metrics if it is initialized successfully

@Yizhou-Yang Is it normal when delete and update data?

Reply:
fixed, now update_before will add metric and return.